### DISCUSSION TITLE
Do notifications methods require `BLUETOOTH_CONNECT` Android permission to be set?

### DISCUSSION THREAD

**Discussion by IlCallo** [capacitor-community/bluetooth-le#636] [Q&A] [answered]
Hi there, I'm developing an app which uses notifications on Android

I noticed that before setting a notification listener or removing one a `writeDescriptor` is performed
It's not clear to me why is that needed, since I already enabled Notify flag only on the descriptor, and I actually have no write permission for the descriptor itself

Right now I just ignored the errors generated by that failed write, since notifications then still work as expected

What I noticed, digging deeper, is that using `writeDescriptor` [require `BLUETOOTH_CONNECT`](https://developer.android.com/reference/android/bluetooth/BluetoothGatt) permission to be set, but this isn't mentioned into this package docs

Is that permission actually needed for notifications to work?
My tests suggest it isn't needed because `writeDescriptor` call isn't actually needed for them to work
Or am I missing something?

**Comment by pwespi** [comment, marked as answer]
Yes, notifications require BLUETOOTH_CONNECT , see https://developer.android.com/reference/android/bluetooth/BluetoothGatt#setCharacteristicNotification(android.bluetooth.BluetoothGattCharacteristic,%20boolean). This plugin request this permission in the initialize call.

**Comment by IlCallo** [comment]
I see, I guess this is the relevant code: https://github.com/capacitor-community/bluetooth-le/blob/main/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/BluetoothLe.kt#L104-L128

But does this mean we don't actually need to add `BLUETOOTH_SCAN`, `ACCESS_FINE_LOCATION`, `ACCESS_COARSE_LOCATION`, etc either?

Cause [here](https://github.com/capacitor-community/bluetooth-le#android) it's mentioned that those must be added into the android manifest, but they're also required into the initialize call
Or I may just be rusty with Android

> I noticed that before setting a notification listener or removing one a writeDescriptor is performed
> It's not clear to me why is that needed, since I already enabled Notify flag only on the descriptor, and I actually have no write permission for the descriptor itself

This question is still bugging me tho
